akka-actor-typed-tests/src/test/java/akka/actor/typed/javadsl/RoutersTest.java

```diff
@@ -18,9 +36,36 @@ public void compileOnlyApiTest() {
         Routers.pool(5, Behaviors.<String>empty()).withRandomRouting().withRoundRobinRouting();
   }
 
-  public void poolBroadcastCompileOnlyApiTest() {
-    Behavior<String> b = Behaviors.receiveMessage((String str) -> Behaviors.same());
+  @Test
+  public void poolBroadcastTest() {
+    TestProbe<String> probe = testKit.createTestProbe();
+    Behavior<String> behavior =
+        Behaviors.receiveMessage(
+            (String str) -> {
+              probe.getRef().tell(str);
+              return Behaviors.same();
+            });
+
     Behavior<String> poolBehavior =
-        Routers.pool(5, b).withBroadcastPredicate(str -> str.startsWith("bc-"));
+        Routers.pool(4, behavior).withBroadcastPredicate(str -> str.startsWith("bc-"));
+
+    ActorRef<String> pool = testKit.spawn(poolBehavior);
+
+    String notBroadcastMsg = "not-bc-message";
+    pool.tell(notBroadcastMsg);
+
+    String broadcastMsg = "bc-message";
+    pool.tell(broadcastMsg);
+
+    List<String> messages = probe.receiveSeveralMessages(5);
+
+    for (int i = 0; i < messages.size(); i++) {
```

Comment:
Oldschool! 

`for (String message : messages) {`

Reply:
Hehe, true. 

The reason is that I need to make a distinction between the first message and the others. But maybe I should have done `receiveMessage` and `receiveSeveralMessages(4)` to avoid this. 

I confess I got some nostalgic feeling. ;-)

Reply:
The code is changed now. Explicitly picking the first message and then the broadcasted ones.